src/SharpLearning.Optimization.Test/ObjectiveUtilities.cs

```diff
@@ -25,5 +26,10 @@ public static OptimizerResult MinimizeWeightFromHeight(double[] parameters)
 
             return new OptimizerResult(parameters, cost);
         }
+
+        public static OptimizerResult MinimizeNonDeterministic(double[] x)
```

Comment:
@jameschch I would move the `Random` member to be an argument for the `MinimizeNonDeterministic` method, That way the seed can be specified instead of hardcoded.

Reply:
This would mean changing the signature of the minimize function. Can you be more specific how you can see this changing?

Reply:
@jameschch Yes, my suggestion is to change the signature of the `MinimizeNonDeterministic` method to include the `Random` as an argument.

Then you will call the optimizer in the test in this way instead, and the test code can decide the seed:

```csharp
var random = new Random(42);
var results = sut.Optimize(p => MinimizeNonDeterministic(p, random));
```